### ISSUE TITLE
Swift's flow.collect function only receives first value

### ISSUE THREAD

**Issue by mattmook** [Kotlin/kotlinx.coroutines#2306]
Having created a simple flow in common code to emit incrementing values with a delay as follows:

```
flow {
    (1..10).forEach {
        emit(it.toString())
        delay(1000)
    }
}.flowOn(Dispatchers.Default)
```

When observing using Swift with:

```
class AModel: Kotlinx_coroutines_coreFlowCollector {    
    func emit(value: Any?, completionHandler: @escaping (KotlinUnit?, Error?) -> Void) {
        print("ios received " + (value as! String))
    }
}

val state: AModel = AModel()

flow.collect(collector: state) { (result: KotlinUnit?, error: Error?) in
    print("completion")
}
```

**Actual:**
Only the first value is received and printed by the Kotlinx_coroutines_coreFlowCollector. Additionally the flows completion block is not executed. The same common code works on Android.

**Expected:**
All values to be received and printed and the flow to complete once the last value has been received.

Kotlin: 1.4.10
Coroutines: 1.3.9-native-mt-2
Xcode: 12.0.1
iOS simulator: iPhone 11 running 14.0
Swift: 5

FYI @qwwdfsad this is the issue we discussed in the Kotlin 1.4 Online booth call.

**Comment by mattmook** [comment]
Sorry, thought I'd uploaded a sample app with this bug report... hopefully it works this time.

[SwiftFlow.zip](https://github.com/Kotlin/kotlinx.coroutines/files/5411381/SwiftFlow.zip)

**Comment by mattmook** [comment]
Retested and still fails with:

Kotlin: 1.4.21
Coroutines: 1.4.2-native-mt

**Comment by ciffa** [comment]
Could it be that you're not calling the completion handler and therefore not clearing the queue?
```
func emit(value: Any?, completionHandler: @escaping (KotlinUnit?, Error?) -> Void) {
    print("ios received " + (value as! String))
    completionHandler(KotlinUnit(), nil)
}
```

**Comment by elevenetc** [comment]
Tested recently released `1.4.3-native-mt` and the issue is resolved.

**Comment by mattmook** [comment]
Yes @ciffa calling the completionHandler does make this work
